source/fundamentals/connection/connection-options.txt

```diff
@@ -6,203 +6,293 @@ Connection Options
-       concurrently. 
+   * - **proxyPort**
+     - non-negative integer
+     - ``null``
+     - Specifies the port of the SOCKS5 proxy server specified in ``proxyHost``.
 
-   * - **maxIdleTimeMS**
-     - integer
-     - ``∞``
-     - Specifies the maximum amount of time a connection can remain idle in the
-       connection pool before being removed and closed.
+   * - **proxyUsername**
+     - string
+     - ``null``
+     - Specifies the username for username/password authentication to the SOCKS5
+       proxy server specified in ``proxyHost``.
 
-   * - **maxPoolSize**
-     - integer
-     - ``100``
-     - Specifies the maximum number of connections that a connection pool may
-       have at a given time.
+   * - **proxyPassword**
+     - string
+     - ``null``
+     - Specifies the password for username/password authentication to the SOCKS5
+       proxy server specified in ``proxyHost``.
 
-   * - **minPoolSize**
-     - integer
-     - ``0``
-     - Specifies the minimum number of connections that must exist at any moment
-       in a single connection pool.
+   * - **readConcernLevel**
+     - string
+     - ``null``
+     - Specifies the default read concern for the client. See :ref:`read concern
+       <read-concern>` for more information.
 
-   * - **noDelay**
-     - boolean
-     - ``true``
-     - Specifies whether to use the TCP socket no-delay option. For more
-       information, see the documentation for `Node.js socket.setNoDelay
-       <https://nodejs.org/dist/latest-v10.x/docs/api/net.html#net_socket_setnodelay_nodelay>`_.
+   * - **readPreference**
+     - string
+     - ``"primary"``
+     - Specifies the default read preference for the client (excluding tags). See :ref:`read preference <read-preference>` for more information.
 
-   * - **pkFactory**
-     - object
+   * - **readPreferenceTags**
+     - comma-separated key:value pairs (e.g. "dc:ny,rack:1" and "dc:ny)
+       can be specified multiple times, each instance of this key is a
+       separate tag set
      - ``null``
-     - Specifies a primary key factory object that generates custom ``_id``
-       keys.
+     - Specifies the default read preference tags for the client. This option is
+       valid only if the read preference mode is not primary.
 
-   * - **promiseLibrary**
-     - object
+       The driver uses the order of the tags in the URI as the order
+       for the read preference.
+
+   * - **replicaSet**
+     - string
      - ``null``
-     - Specifies the Promise library class the application uses (e.g. Bluebird).
-       This library must be compatible with ES6.
+     - Specifies the name of the replica set to connect to.
 
-   * - **promoteBuffers**
+   * - **retryReads**
      - boolean
-     - ``false``
-     - Specifies whether to promote Binary BSON values to native Node.js
-       ``Buffer`` type data.
+     - ``true``
+     - Enables retryable reads on MongoDB version 3.6 or later.
```

Comment:
I'd be tempted not to mention 3.6 here as our 4.x driver does not support any server below version 3.6..